### PR TITLE
[charts] Update README.md

### PR DESCRIPTION
**Pull request by alexfauquette**
- Update the readme
- Remove the `d3-array` package

### CODE REVIEW

**Review by LukasTy — APPROVED** [review]
🙌

File: packages/x-charts/README.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
-# @mui/x-charts
+# @mui/charts
```
Comment (alexfauquette) [inline packages/x-charts/README.md:1]:
```suggestion
# @mui/x-charts
```

Comment (LukasTy) [reply packages/x-charts/README.md]: 🤦 🙈 💯

File: packages/x-charts/README.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+    "@mui/material": "^5.4.1",
+    "@mui/system": "^5.4.1",
+    "react": "^17.0.0 || ^18.0.0",
+    "react-dom": "^17.0.0 || ^18.0.0"
+},
+```
+
+## Documentation
+
+[The documentation](https://mui.com/x/react-charts)
```
Comment (oliviertassinari) [inline packages/x-charts/README.md:33]: Fixed in https://github.com/mui/mui-x/commit/db1a54b5626c68db56ddf848a97e5f21c3427b39

File: packages/x-charts/README.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+
+```sh
+yarn add @mui/x-charts
+```
+
+This component has the following peer dependencies that you will need to install as well.
+
+```json
+"peerDependencies": {
+    "@mui/material": "^5.4.1",
```
Comment (oliviertassinari) [inline packages/x-charts/README.md:24]: I think that a peer dependency will be a major blocker for the broader adoption of the chart library. It's OK in the near future, but I think that we need to have a plan for how to make it optional.

Comment (LukasTy) [reply packages/x-charts/README.md:24]: That is a great observation! 👌 
It looks like we mainly depend on the styling system, which can easily be replaced with imports from the `@mui/system`.
But there are a few usages of actual `material` imports like `Popper`, `Paper`, and `NoSsr`, which shouldn't be too difficult to replace. 
cc @alexfauquette

**Pull request merged**